Prevent checkout step heading text overlapping actual heading on small viewports

opr opened the pull request:
This change stops the `wc-block-components-checkout-step__heading-content` class being positioned absolutely, which caused it to overlap the checkout step title on smaller viewports. The heading content now drops below the heading when the text will overlap. It also adds the same grey line to the left of the heading content as on the checkout step description.

Fixes #3082

⚠️ Because Safari does not support the CSS property `gap` the text sits very close to the heading. There is not an easy way to set the margins on this text without specifically targeting Safari, however `gap` will be supported in the next version of Safari according to https://caniuse.com/flexbox-gap

### Screenshots

#### Before
![image](https://user-images.githubusercontent.com/5656702/99527687-1b4f9b80-2995-11eb-8972-092e522f6190.png)

#### After
<img width="336" alt="Screenshot 2020-11-18 at 11 56 56" src="https://user-images.githubusercontent.com/5656702/99527717-2dc9d500-2995-11eb-9dec-694e65ade2e3.png">

#### In Safari
<img width="357" alt="Screenshot 2020-11-18 at 12 05 09" src="https://user-images.githubusercontent.com/5656702/99528517-51414f80-2996-11eb-895b-6d56f6552024.png">

### How to test the changes in this Pull Request:

1. Ensure the Allow customers to log into an existing account during checkout option is enabled in **WooCommerce -> Settings -> Accounts & Privacy**.
2. Log out of the site and add a product to your cart.
3. Go to a page with the checkout block on and reduce the viewport so far that the "Already have an account? Log in" text is about to overlap.
4. Ensure the text drops to below the header and does not overlap with the section heading.

<!-- If you can, add the appropriate labels -->

opr committed:
Change style rules for checkout step headings

This is required because when the checkout heading text is too long it collided and overlapped the heading content due to its absolute position. In this commit we add a grey line to the left of the checkout step heading to ensure consistency with the checkout step container's styling.

mikejolley commented on assets/js/base/components/cart-checkout/form-step/style.scss:
```diff
@@ -12,6 +12,16 @@
 		padding-right: $gap-large;
 	}
+
+	.wc-block-components-checkout-step__heading::after {
```
I wasn't sure what this was for but I see it's for the border. This part doesn't work in Safari—expected?

opr replied:
This creates a short grey line at the bottom of the `step__heading` - it _is_ present in Safari, but because `gap` is not supported it does not line up well, so not immediately obvious that it is there. I think we opt to leave this in, because when Safari supports `gap` it'll line up automatically and have the grey border next to the "Already have an account" text as it does in Chrome and Firefox.

mikejolley commented on assets/js/base/components/cart-checkout/form-step/style.scss:
```diff
@@ -45,8 +57,7 @@
 	@include font-size(smaller);
-	position: absolute;
-	right: 0;
+	position: relative;
```
Is position needed at all?

opr replied:
Good catch, no it isn't!

mikejolley reviewed (approved):
👍🏻  I tested this and it worked as described.

As you said, Safari doesn't support this so it looks like this:
![Checkout – one wordpress test 2020-11-19 11-26-27](https://user-images.githubusercontent.com/90977/99661207-6a600400-2a5b-11eb-93b4-beff93535c18.png)

It's still acceptable and better than overlap through :)

Firefox also looked ok: 

![Checkout – one wordpress test 2020-11-19 11-28-22](https://user-images.githubusercontent.com/90977/99661290-89f72c80-2a5b-11eb-99f5-d4f7eabf89d6.png)

Curious as to why Chrome shows these fonts much larger than Safari and FF...

Anyhow, added a small note but seems good to merge.

As an aside (@nerrad) this link could be removed entirely if we prioritised https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2751-- the link text could go below email and trigger a password field to appear so it's all inline. Would be a better experience :)

nerrad commented:
> As an aside (@nerrad) this link could be removed entirely if we prioritised #2751 the link text could go below email and trigger a password field to appear so it's all inline. Would be a better experience :)

Agreed. It should surface in one of the weekly triages product and myself are doing.

opr commented:
@mikejolley The font sizes are the same for me on Chrome and Firefox, although it looks like Firefox renders the text a little bit heavier. This is in TwentyTwenty theme.

### Chrome
<img width="301" alt="chrome" src="https://user-images.githubusercontent.com/5656702/99704757-b7120200-2a90-11eb-84be-fddf0022c2d2.png">

### Firefox
<img width="327" alt="ff" src="https://user-images.githubusercontent.com/5656702/99704777-be391000-2a90-11eb-8c24-84241250e786.png">

opr committed:
Remove position rule from checkout-step heading content

This was no longer needed as it nothing relied on the position of this element.
